Startup.cs

```diff
@@ -68,16 +69,34 @@ public void ConfigureServices(IServiceCollection services)
                 options.ResponseType = "id_token code";
                 options.Scope.Add("offline_access");
                 options.Scope.Add("User.Read");
-//                options.Prompt = "consent";
-                
-                // Handling the auth code
+                //                options.Prompt = "consent";
+
+                // Handling the auth redemption by MSAL.NET so that a token is available in the token cache
+                // where it will be usable from Controllers later (through the TokenAcquisition service)
                 var handler = options.Events.OnAuthorizationCodeReceived;
                 options.Events.OnAuthorizationCodeReceived = async context =>
                 {
                     var _tokenAcquisition = context.HttpContext.RequestServices.GetRequiredService<ITokenAcquisition>();
                     await _tokenAcquisition.AddAccountToCacheFromAuthorizationCode(context, options.Scope);
                     await handler(context);
                 };
+
+                // Avoids having users being presented the select account dialog when they are already signed-in
+                // for instance when going through incremental consent 
+                options.Events.OnRedirectToIdentityProvider = async context =>
+                {
+                    string login = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.LoginHint);
+                    if (!string.IsNullOrWhiteSpace(login))
+                    {
+                        context.ProtocolMessage.LoginHint = login;
+                        context.ProtocolMessage.DomainHint = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.DomainHint);
```

Comment:
Are you sure this is going to be set? I know you set it yourself, but can the LoginHint be set from somewhere else, without setting the domain?

Reply:
@bgavrilMS : the domain hint is not the domain but a hint helping the STS to bypass the account selection by directing directly to AAD or MSA.
I don't understand your question "are you sure this is going to be set?". It's set in the Controller's methods in `BuildAuthenticationPropertiesForIncrementalConsent` when the identity of the sign-ed in user is known (might not be the case in `OnRedirectToIdentityProvider`